src/windows_native/descriptor/mod.rs

```diff
@@ -0,0 +1,415 @@
+        let mut last_bit_position: HashMap<(MainItems, u8), i32> = HashMap::new();
+        let mut last_report_item_lookup: HashMap<(MainItems, u8), usize> = HashMap::new();
+
+        let mut index = 0;
+        while index < main_item_list.len() {
+            let current = main_item_list[index];
+            if ReportType::try_from(current.main_item_type).is_ok() {
+                let lbp = last_bit_position
+                    .get(&(current.main_item_type, current.report_id))
+                    .copied()
+                    .unwrap_or(-1);
+                let lrip = last_report_item_lookup
+                    .get(&(current.main_item_type, current.report_id))
+                    .copied();
+                if lbp + 1 != current.first_bit as i32 && lrip.is_some_and(|i|
+                        main_item_list[i].first_bit != current.first_bit) {
+                    let list_node = search_list(lbp, current.main_item_type, current.report_id, lrip.unwrap(), &main_item_list);
+                    main_item_list.insert(list_node + 1, MainItemNode::new((lbp + 1) as u16, current.first_bit - 1, ItemNodeType::Padding, -1, 0, current.main_item_type, current.report_id));
+                    index += 1;
+                }
+                last_bit_position.insert((current.main_item_type, current.report_id), current.last_bit as i32);
+                last_report_item_lookup.insert((current.main_item_type, current.report_id), index);
+            }
+            index += 1;
+        }
+
+        for rt_idx in ReportType::values() {
+            for report_idx in 0..=255 {
+                if let Some(lbp) = last_bit_position.get(&(rt_idx.into(), report_idx)) {
+                    let padding = 8 - ((*lbp + 1) % 8);
+                    if padding < 8 {
+                        // Insert padding item after item referenced in last_report_item_lookup
+                        let lrip = *last_report_item_lookup.get(&(rt_idx.into(), report_idx)).unwrap();
+                        main_item_list.insert(lrip + 1, MainItemNode::new((lbp + 1) as u16, (lbp + padding) as u16, ItemNodeType::Padding, -1, 0, rt_idx.into(), report_idx));
+                        last_report_item_lookup
+                            .values_mut()
+                            .filter(|i| **i > lrip)
+                            .for_each(|i| *i += 1);
+                    }
+                }
+            }
+        }
+    }
+    main_item_list
+}
+
+fn search_list(search_bit: i32, main_item_type: MainItems, report_id: u8, start: usize, list: &[MainItemNode]) -> usize {
+    list[start..]
+        .iter()
+        .peaking()
+        .position(|(_, next)| next.is_some_and(|next| next.main_item_type == MainItems::Collection ||
+           next.main_item_type == MainItems::CollectionEnd ||
+           (next.last_bit as i32 >= search_bit && next.report_id == report_id && next.main_item_type == main_item_type) ))
+        .unwrap() + start
+}
+
+
+trait PeakIterExt<T: Iterator> {
```

Comment:
Maybe put the PeakIterExt utility and related code into its own file (still in  the windows backend).
It might be used from more code in the future, so it`s nice to encapsulate in its own module.